finetuner/tuner/evaluation.py

```diff
@@ -0,0 +1,160 @@
+from typing import TYPE_CHECKING, Optional, List, Dict, Tuple
+from jina import Document, DocumentArray
+from jina.math.evaluation import (
+    r_precision,
+    precision_at_k,
+    recall_at_k,
+    f1_score_at_k,
+    hit_at_k,
+    average_precision,
+    reciprocal_rank,
+    dcg_at_k,
+    ndcg_at_k,
+)
+
+from .. import __default_tag_key__
+from ..embedding import embed
+
+if TYPE_CHECKING:
+    from ..helper import AnyDNN, DocumentSequence
+
+
+METRICS = {
+    "r_precision": r_precision,
+    "precision": precision_at_k,
+    "recall": recall_at_k,
+    "f1score": f1_score_at_k,
+    "average_precision": average_precision,
+    "hit": hit_at_k,
+    "reciprocal_rank": reciprocal_rank,
+    "dcg": dcg_at_k,
+    "ndcg": ndcg_at_k,
+}
+
+
+class Evaluator:
+    def __init__(
+        self,
+        eval_data: 'DocumentSequence',
+        catalog: 'DocumentSequence',
+        embed_model: Optional['AnyDNN'] = None,
+        distance_metric: str = 'cosine',
+        limit: int = 20,
+    ):
+        """
+        Build an Evaluator object that can be used to evaluate an embedding model on a retrieval task
+
+        :param eval_data: a sequence of documents. Each document should contain ground truth matches under
+            ``doc.matches`` and relevance scores under ``doc.tags[__default_tag_key__]['label']``
+        :param catalog: a sequence of documents, against whist the eval docs will be matched.
+        :param embed_model: the embedding model to use, in order to extract document representations.
+        :param distance_metric: which distance metric to use when matching documents
```

Comment:
can you document the allowed metrics here as well?

Reply:
done